pkg/registry/core/pod/strategy.go

```diff
@@ -205,6 +205,18 @@ type podEphemeralContainersStrategy struct {
 // EphemeralContainersStrategy wraps and exports the used podStrategy for the storage package.
 var EphemeralContainersStrategy = podEphemeralContainersStrategy{Strategy}
 
+func (podEphemeralContainersStrategy) PrepareForUpdate(ctx context.Context, obj, old runtime.Object) {
+	newPod := obj.(*api.Pod)
+	oldPod := old.(*api.Pod)
+
+	// Drop all changes except for pod.Spec.EphemeralContainers
+	ec := newPod.Spec.EphemeralContainers
+	*newPod = *oldPod
```

Comment:
comment on why we're shallow copying... I think that's sufficient to avoid mutating the original for all the fields we're preserving, but DeepCopy is more robust against future changes that preserve more fields a shallow-copy wouldn't make safe.